jms/src/main/scala/akka/stream/alpakka/jms/JmsProducerStage.scala

```diff
@@ -100,41 +110,82 @@ private[jms] final class JmsProducerStage[A <: JmsMessage, PassThrough](settings
         }
       )
 
+      override def onTimer(timerKey: Any): Unit = timerKey match {
+        case s: SendAttempt[E] => sendWithRetries(s)
+        case _ => ()
+      }
+
+      private def sendWithRetries(send: SendAttempt[E]): Unit = {
+        import send._
+        val eventualSend = if (jmsProducers.nonEmpty) {
+          val jmsProducer: JmsMessageProducer = jmsProducers.dequeue()
+          Future(jmsProducer.send(message.message)).andThen {
+            case tried => sendCompletedCB.invoke((send, tried, jmsProducer))
+          }
+        } else {
+          nextTryOrFail(new RuntimeException("JmsProducer is not connected, skipping send attempt"), send)
```

Comment:
Potentially, the send attempt can happen while the producer is disconnected. Here, this is handled as a regular attempt that counts towards the max retries.